x-pack/platform/plugins/shared/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -428,17 +428,6 @@ export function buildComponentTemplates(params: {
       isSyntheticSourceEnabledByDefault ||
       isTimeSeriesEnabledByDefault);
 
-  const isPkgConfiguringDynamicSettings =
-    Object.keys(mappingsRuntimeFields).length > 0 || indexTemplateMappings?.dynamic !== undefined;
-
-  // Setting overrides for otel input packages, but only if the packages don't explicitly disable dynamic mappings or use `dynamic: runtime`
-  // Override the `dynamic: false` set in otel@mappings to avoid conflicts in case
-  const shouldOverrideSettingsForOtelInputs = isOtelInputType && !isPkgConfiguringDynamicSettings;
-
-  // Override `subobjects: false` to avoid conflicts with traces-otel@mappings
-  const shouldOverrideSettingsForOtelInputsTraces =
-    shouldOverrideSettingsForOtelInputs && type === 'traces' && !indexTemplateMappings.runtime;
```

Comment:
These overrides can be avoided now that non-otel mappings are removed, and packages are not required to define fields.